src/components/AddTapModal.js

```diff
@@ -0,0 +1,599 @@
+                        type="radio"
+                        label="Yes"
+                        name="VesselRadios"
+                        id="VesselRadios1"
+                        value="yes"
+                      />
+
+                      <Form.Check
+                        className="radioText"
+                        type="radio"
+                        label="No"
+                        name="VesselRadios"
+                        id="VesselRadios2"
+                        value="no"
+                      />
+                    </Form.Group>
+
+                    <Form.Group
+                      controlId="Phlask Statement"
+                      value={this.state.phlaskStatement}
+                      onChange={this.onChangePhlaskStatement}
+                    >
+                      <Form.Label>
+                        <strong>PHLASK Statement</strong>
+                      </Form.Label>
+                      <Form.Control
+                        as="textarea"
+                        rows="2"
+                        placeholder="Please use this section to make any statement about your organization or enterprise!"
+                      />
+                    </Form.Group>
+
+                    <Form.Group
+                      controlId="Norms and Rules"
+                      value={this.state.normsAndRules}
+                      onChange={this.onChangeNormsAndRules}
+                    >
+                      <Form.Label>
+                        <strong>Norms and Rules</strong>
+                      </Form.Label>
+                      <Form.Control
+                        as="textarea"
+                        rows="3"
+                        placeholder="PHLASKing is intended to be an unobtrusive part of doing business. If there are special norms associated with accessing water, please use this space to describe them."
+                      />
+                    </Form.Group>
+                  </div>
+                </Accordion.Collapse>
+              </Accordion>
+
+              <ImageUploader
+                withIcon={true}
+                buttonText="Choose images"
+                onChange={this.onDrop}
+                imgExtension={[".jpg", ".gif", ".png", ".gif"]}
+                maxFileSize={5242880}
+                withPreview={true}
+              />
+            </Modal.Body>
+            <Modal.Footer>
+              <Button variant="secondary" onClick={this.handleClose}>
+                Close
+              </Button>
+              <Button variant="primary" type="submit" onClick={this.handleClose}>
+                Submit
+              </Button>
+            </Modal.Footer>
+          </Form>
+        </Modal>
+
+        <button
+          onClick={this.handleShow}
+          className={`${isMobile ? styles.mobileAddButton : ""} ${
+            styles.addButton
+          }`}
+        >
+          <FontAwesomeIcon icon={faPlus} size="2x" />
+        </button>
```

Comment:
I put the button and all of the display logic into the component so I could just drop it into the Toolbar without modifying anything else.